internal/controllers/topology/cluster/desired_state.go

```diff
@@ -870,6 +887,281 @@ func isMachineDeploymentDeferred(clusterTopology *clusterv1.Topology, mdTopology
+func computeMachinePool(_ context.Context, s *scope.Scope, machinePoolTopology clusterv1.MachinePoolTopology) (*scope.MachinePoolState, error) {
+	desiredMachinePool := &scope.MachinePoolState{}
+
+	// Gets the blueprint for the MachinePool class.
+	className := machinePoolTopology.Class
+	machinePoolBlueprint, ok := s.Blueprint.MachinePools[className]
+	if !ok {
+		return nil, errors.Errorf("MachinePool class %s not found in %s", className, tlog.KObj{Obj: s.Blueprint.ClusterClass})
+	}
+
+	var machinePoolClass *clusterv1.MachinePoolClass
+	for _, mpClass := range s.Blueprint.ClusterClass.Spec.Workers.MachinePools {
+		mpClass := mpClass
+		if mpClass.Class == className {
+			machinePoolClass = &mpClass
+			break
+		}
+	}
+	if machinePoolClass == nil {
+		return nil, errors.Errorf("MachinePool class %s not found in %s", className, tlog.KObj{Obj: s.Blueprint.ClusterClass})
+	}
+
+	// Compute the bootstrap template.
+	currentMachinePool := s.Current.MachinePools[machinePoolTopology.Name]
+	var currentBootstrapConfigRef *corev1.ObjectReference
+	if currentMachinePool != nil && currentMachinePool.BootstrapObject != nil {
+		currentBootstrapConfigRef = currentMachinePool.Object.Spec.Template.Spec.Bootstrap.ConfigRef
+	}
+	var err error
+	desiredMachinePool.BootstrapObject, err = templateToObject(templateToInput{
+		template:              machinePoolBlueprint.BootstrapConfig,
+		templateClonedFromRef: contract.ObjToRef(machinePoolBlueprint.BootstrapConfig),
+		cluster:               s.Current.Cluster,
+		namePrefix:            bootstrapTemplateNamePrefix(s.Current.Cluster.Name, machinePoolTopology.Name),
+		currentObjectRef:      currentBootstrapConfigRef,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to compute bootstrap object for topology %q", machinePoolTopology.Name)
+	}
+
+	bootstrapTemplateLabels := desiredMachinePool.BootstrapObject.GetLabels()
+	if bootstrapTemplateLabels == nil {
+		bootstrapTemplateLabels = map[string]string{}
+	}
+	// Add ClusterTopologyMachinePoolLabel to the generated Bootstrap template
+	bootstrapTemplateLabels[clusterv1.ClusterTopologyMachinePoolNameLabel] = machinePoolTopology.Name
+	desiredMachinePool.BootstrapObject.SetLabels(bootstrapTemplateLabels)
+
+	// Compute the Infrastructure template.
+	var currentInfraMachineTemplateRef *corev1.ObjectReference
+	if currentMachinePool != nil && currentMachinePool.InfrastructureMachinePoolObject != nil {
+		currentInfraMachineTemplateRef = &currentMachinePool.Object.Spec.Template.Spec.InfrastructureRef
+	}
+	desiredMachinePool.InfrastructureMachinePoolObject, err = templateToObject(templateToInput{
+		template:              machinePoolBlueprint.InfrastructureMachinePoolTemplate,
+		templateClonedFromRef: contract.ObjToRef(machinePoolBlueprint.InfrastructureMachinePoolTemplate),
+		cluster:               s.Current.Cluster,
+		namePrefix:            infrastructureMachineTemplateNamePrefix(s.Current.Cluster.Name, machinePoolTopology.Name),
```

Comment:
```suggestion
		namePrefix:            infrastructureMachinePoolNamePrefix(s.Current.Cluster.Name, machinePoolTopology.Name),
```
let's create a new func (with the same calculation)